crates/ide-assists/src/handlers/sort_items.rs

```diff
@@ -121,24 +122,33 @@ trait AddRewrite {
 impl AddRewrite for Assists {
     fn add_rewrite<T: AstNode>(
         &mut self,
+        ctx: &AssistContext<'_>,
         label: &str,
         old: Vec<T>,
         new: Vec<T>,
         target: TextRange,
     ) -> Option<()> {
+        let node = old.first().unwrap().syntax().parent().unwrap();
         self.add(AssistId("sort_items", AssistKind::RefactorRewrite), label, target, |builder| {
-            let mutable: Vec<T> = old.into_iter().map(|it| builder.make_mut(it)).collect();
-            mutable
-                .into_iter()
-                .zip(new)
-                .for_each(|(old, new)| ted::replace(old.syntax(), new.clone_for_update().syntax()));
+            let mut editor = builder.make_editor(&node);
+
+            old.into_iter().zip(new).for_each(|(old, new)| {
+                // FIXME: remove `clone_for_update` when `SyntaxEditor` handles it for us
+                editor.replace(old.syntax(), new.clone_for_update().syntax())
+            });
+
+            builder.add_file_edits(ctx.file_id(), editor)
```

Comment:
I think we can get around passing the `ctx` everywhere but just using the `file_id` from the builder:
```suggestion
            builder.add_file_edits(builder.fild_id, editor)
```

Reply:
Right, updated